contracts/src/PDPRecordKeeperApplication.sol

```diff
@@ -1,36 +1,47 @@
 // SPDX-License-Identifier: UNLICENSED
 pragma solidity ^0.8.13;
 
-// PDPRecordKeeper is a default implementation of a record keeper for the PDP service.
-// It maintains a record of all events that have occurred in the PDP service,
-// and provides a way to query these events.
-// This contract only supports one PDP service caller, set in the constructor.
-contract PDPRecordKeeper {
-    // The address of the PDP service contract that is allowed to call this contract
-    address public immutable pdpServiceAddress;
+import {PDPService} from "./PDPService.sol";
 
-    // Enum to represent different types of operations
+interface PDPApplication {
     enum OperationType {
         NONE,
         CREATE,
         ADD,
-        REMOVE,
+        REMOVE_SCHEDULED,
```

Comment:
I think we need to hear about removal schedule otherwise the listener can't fail on removal without breaking liveness

Reply:
For example imagining an application like f05 as the listener you'd want to make sure that you are allowed to remove a cid from the proofset because the deal is expired.

Reply:
Oooh, we're running into similar issues as the sector activation event. I hadn't thought about the listener failing and being able to prevent the SP from performing some operations. My first thought is that the PDP service should catch and ignore listener failures. SLA enforcement in this way wasn't the goal. That's a good discussion for us this week.